Platform/NXP/Sabre_iMX6SX_1GB/Drivers/GraphicsOutputDxe/GraphicsOutputDxe.c

```diff
@@ -0,0 +1,415 @@
+    PreferredTiming->PixelRepetition = 1;
+    PreferredTiming->PixelFormat = PIXEL_FORMAT_BGRA32;
+    PreferredTiming->Bpp = 32;
+
+    ImxPrintDisplayTiming(
+        "Preferred Timing",
+        PreferredTiming);
+
+    return;
+}
+
+//
+// Entry Point
+//
+EFI_STATUS
+GraphicsOutputDxeInitialize(
+    IN EFI_HANDLE           ImageHandle,
+    IN EFI_SYSTEM_TABLE *   SystemTable
+    )
+{
+    EFI_STATUS status;
+    ZeroMem(&sgInst, sizeof(sgInst));
+
+    DEBUG((DEBUG_INFO,"GraphicsOutputDxeInitialize\n"));
+
+    // Figure out what the preferred timing
+    GetPreferredTiming(&sgInst.PreferredTiming);
+
+    // Allocate frame buffer
+    UINT32 fbSize =
+        sgInst.PreferredTiming.HActive *
+        sgInst.PreferredTiming.VActive *
+        (sgInst.PreferredTiming.Bpp / 8);
+
+    DebugPrint(DEBUG_ERROR, "FbSize = %x\n", fbSize);
+    if (fbSize > FixedPcdGet32(PcdFrameBufferSize)) {
+        DebugPrint(DEBUG_ERROR, "Requested Frame buffer size too big for reserved region\n");
+        status = EFI_OUT_OF_RESOURCES;
+        goto Exit;
+    }
+
+    sgInst.FrameBuffer = (VOID*)FixedPcdGet32(PcdFrameBufferBase);
+    DebugPrint(DEBUG_ERROR, "Framebuffer = %x\n", sgInst.FrameBuffer);
+
+    // Initialize the frame buffer to black
+    SetMem32(
+        sgInst.FrameBuffer,
+        fbSize,
+        0xFF000000);
+
+    status = ConfigureLCD(
+        &sgInst.PreferredTiming,
+        sgInst.FrameBuffer);
+    if (EFI_ERROR(status)) {
+        DebugPrint(DEBUG_ERROR, "GraphicsOutputDxeInitialize - ConfigureLCD failed\n");
+        goto Exit;
+    }
+
+    // Graphics output protocol
+    sgInst.Protocol.QueryMode = sGfxQueryMode;
+    sgInst.Protocol.SetMode = sGfxSetMode;
+    sgInst.Protocol.Blt = sGfxBlt;
+    sgInst.Protocol.Mode = &sgInst.Mode;
+
+    sgInst.ModeInfo.Version = 0;
+    sgInst.ModeInfo.HorizontalResolution = sgInst.PreferredTiming.HActive;
+    sgInst.ModeInfo.VerticalResolution = sgInst.PreferredTiming.VActive;
+    sgInst.ModeInfo.PixelFormat = PixelBlueGreenRedReserved8BitPerColor;
+    sgInst.ModeInfo.PixelsPerScanLine = sgInst.ModeInfo.HorizontalResolution;
+
+    sgInst.Mode.MaxMode = 1;
+    sgInst.Mode.Mode = 0;
+    sgInst.Mode.Info = &sgInst.ModeInfo;
+    sgInst.Mode.SizeOfInfo = sizeof(EFI_GRAPHICS_OUTPUT_MODE_INFORMATION);
+    sgInst.Mode.FrameBufferBase = (UINT32)sgInst.FrameBuffer;
+    sgInst.Mode.FrameBufferSize = sgInst.ModeInfo.HorizontalResolution * sgInst.ModeInfo.VerticalResolution * (sgInst.PreferredTiming.Bpp / 8);
+
+    // Install
+    status = gBS->InstallMultipleProtocolInterfaces(&ImageHandle,
+        &gEfiGraphicsOutputProtocolGuid, &sgInst.Protocol,
+        &gEfiDevicePathProtocolGuid, &g_VidDevicePath,
+        NULL);
+
+Exit:
+    return status;
+}
+
+// -------------------------------------------------------------------------------------------------------
+
+// Almost Identical to iMXVideoDxe.c VidGopBlt
+static
+EFI_STATUS
+sGfxBlt(
```

Comment:
We recently introduced library code into iMXDisplayLib to implement Blt, QueryMode, and SetMode (https://github.com/ms-iot/imx-edk2-platforms/blob/imx/Silicon/NXP/iMXPlatformPkg/Library/iMXDisplayLib). I recommend using that library to reduce code copy/bloat.

See https://github.com/ms-iot/imx-edk2-platforms/pull/30 and https://github.com/ms-iot/imx-edk2-platforms/blame/imx/Platform/Udoo/UdooNeo_iMX6SX_1GB/Drivers/GraphicsOutputDxe/GraphicsOutputDxe.c#L272-L274 for examples of how to use this lib.

Reply:
Driver updated.